pkg/resource/fake/mocks.go

```diff
@@ -157,18 +149,61 @@ func (m *CredentialsSecretReferencer) GetCredentialsSecretReference() v1alpha1.S
 	return m.Ref
 }
 
+// TODO(negz): Remove WorkloadReferencer? Is it still used anywhere?
```

Comment:
@negz confirmed we can remove, would you mind going ahead and taking it out in this PR? ref: https://github.com/crossplane/oam-kubernetes-runtime/blob/0be8044c8384876133cb48fa33be4f46a1296694/pkg/oam/oam.go#L44